### DISCUSSION TITLE
Update from v24.8.5 not possible.

### DISCUSSION THREAD

**Discussion by svenbrose** [tbnobody/OpenDTU#2327] [Q&A]
Update from v24.8.5 is not possible. 
Tried the newest and the next (24.9.20), opendtu-generic.bin and opendtu-generic_esp32.bin.

Upgrade runs without errormessages but the version is still on v24.8.5

Any ideas?

Thank you!
![Greenshot 2024-10-03 21 24 43](https://github.com/user-attachments/assets/b5955ef5-71f1-4694-ae65-113062ea5b4e)

**Comment by tbnobody** [comment]
could you please doublecheck whether you are logged-in on the web UI before flashing?

**Reply by svenbrose** [reply]
Yes. You cannot reach the site without be logged in.
Tried it in Safari and Chrome.

**Reply by ghost** [reply]
Hi,
hat es mittlerweile geklappt?
Mal eine blöde Frage: Wie hast du die aktuelle Firmware ausgewählt? Wenn du den Links in der openDTU Oberfläche (dein Foto) folgst, landest du im Release der momentan installierten Firmware. Mir ist es auch schon 2 - 3 mal passiert, das ich diese fälschlicherweise eingespielt habe, dann ändert sich natürlich nichts. Die aktuelle Soft findet sich unter [Releases](https://github.com/tbnobody/OpenDTU/releases) ganz oben oder [hier (24.10.2)](https://github.com/tbnobody/OpenDTU/releases/tag/v24.10.2).

**Reply by svenbrose** [reply]
> Hi, hat es mittlerweile geklappt? Mal eine blöde Frage: Wie hast du die aktuelle Firmware ausgewählt? Wenn du den Links in der openDTU Oberfläche (dein Foto) folgst, landest du im Release der momentan installierten Firmware. Mir ist es auch schon 2 - 3 mal passiert, das ich diese fälschlicherweise eingespielt habe, dann ändert sich natürlich nichts. Die aktuelle Soft findet sich unter [Releases](https://github.com/tbnobody/OpenDTU/releases) ganz oben oder [hier (24.10.2)](https://github.com/tbnobody/OpenDTU/releases/tag/v24.10.2).

Leider nicht.
Sorry, für die späte Antwort. War 3 Tage unterwegs. 
Ich klicke auf "Firmware Version / Git Hash" (die aktuelle Version v24.8.5  bei mir). Lande auf https://github.com/tbnobody/OpenDTU/releases/tag/v24.8.5
Dort klicke ich auf "Releases" und lande auf https://github.com/tbnobody/OpenDTU/releases. Dort steht v24.10.6.
Lade opendtu-generic.bin oder opendtu-generic_esp32.bin (probiere immer beide).

**Reply by ghost** [reply]
Das ist schon komisch.

Da es gerade eine neue Firmware gibt habe ich das genau so durchgespielt: 
- Über Firmware Version / Git Hash zu [v24.10.6](https://github.com/tbnobody/OpenDTU/releases/tag/v24.10.6)
- [Releases](https://github.com/tbnobody/OpenDTU/releases)
- [v24.10.15](https://github.com/tbnobody/OpenDTU/releases/tag/v24.10.15)
- [opendtu-generic_esp32.bin](https://github.com/tbnobody/OpenDTU/releases/download/v24.10.15/opendtu-generic_esp32.bin)
Letztere OTA zur DTU geschickt; damit ist eine meiner DTUs sofort wieder mit v24.10.15 aktuell.

Empfohlen wird wohl die opendtu-generic.bin zu meiden und besser die .....esp32.bin zu verwenden.

Hmm, was könnte noch helfen?Ich philosophiere mal ein wenig:
Evtl. spielt dir dein Browser einen Streich. Mal alle Browserdaten löschen, bzw. gelegentlich F5 drücken. Chrome zeigt bei mir die Adresse der Links korrekt bzw. passend an. Im Link ist die Version mit drin.
Die Verlinkungen in meinem Beitrag sollten auf jeden Fall dazu führen, hier die richtige Dateiversion zu erhalten (v24.10.15)

Alternativ: _Rechtsklick_, _Link speichern unter_ und einen neuen Dateinamen vergeben und diese Datei Flashen.

Oder: Du könntest die [opendtu-generic_esp32.factory.bin](https://github.com/tbnobody/OpenDTU/releases/download/v24.10.15/opendtu-generic_esp32.factory.bin) mal probieren, allerdings musst du bei dir danach deine Einstellungen wiederherstellen, also diese möglichst vorher in 2 Dateien sichern. Ich kann leider nicht auf die Schnelle sagen, ob die ...factory... per Webupdate geschickt werden darf und der ESP danach normal funktioniert.

Hast du es mal von einem anderen Endgerät versucht? Ein Update geht auch vom smarten Handy aus.
D.h., ich gehe davon aus, du updatest über die OpenDTU Weboberfläche.

Vielleicht will der ESP32 ja platt gemacht werden...
Du könntest ebenso per [USB die DTU aktualisieren.](https://www.opendtu.solar/firmware/)
Ich verwende, wenn es sein muss, den [Espressif Web Flasher](https://espressif.github.io/esptool-js/). Damit kann man den ESP32 auch sehr einfach komplett löschen. Danach ist natürlich die [opendtu-generic_esp32.factory.bin](https://github.com/tbnobody/OpenDTU/releases/download/v24.10.15/opendtu-generic_esp32.factory.bin) an die Startadresse 0x0 des ESP32 zu senden. (Nicht-factory an die Adresse 0x10000)
